plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntitySheared.java

```diff
@@ -0,0 +1,88 @@
+package com.denizenscript.denizen.objects.properties.entity;
+
+import com.denizenscript.denizen.nms.NMSHandler;
+import com.denizenscript.denizen.nms.NMSVersion;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.utilities.BukkitImplDeprecations;
+import com.denizenscript.denizencore.objects.Mechanism;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.properties.PropertyParser;
+import org.bukkit.entity.*;
+
+public class EntitySheared extends EntityProperty<ElementTag> {
+
+    // <--[property]
+    // @object EntityTag
+    // @name sheared
+    // @input ElementTag(Boolean)
+    // @description
+    // Controls whether a sheep is sheared, a bogged is harvested or a snow golem is derped, ie not wearing a pumpkin.
+    // To include drops or harvesting mushroom cows consider <@link mechanism EntityTag.shear>.
+    // -->
+
+    public static boolean describes(EntityTag entity) {
+        return entity.getBukkitEntity() instanceof Sheep
+                || entity.getBukkitEntity() instanceof Snowman
+                || (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21) && entity.getBukkitEntity() instanceof Shearable);
+    }
+
+    @Override
+    public ElementTag getPropertyValue() {
+        return new ElementTag(isSheared());
+    }
+
+    @Override
+    public void setPropertyValue(ElementTag param, Mechanism mechanism) {
+        if (mechanism.requireBoolean()) {
+            setSheared(param.asBoolean());
+        }
+    }
+
+    public boolean isSheared() {
+        if (getEntity() instanceof Sheep sheep) {
+            return sheep.isSheared();
+        }
+        else if (getEntity() instanceof Snowman snowman) {
+            return snowman.isDerp();
+        }
+        else if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_21) && getEntity() instanceof Shearable shearable) {
+            return shearable.isSheared();
+        }
+        return false;
+    }
+
+    public void setSheared(boolean sheared) {
```

Comment:
Any reason to have these in separate methods? not necessarily a problem, but generally the logic is just directly in `get/setPropertyValue` (and e.g. the legacy `is_sheared` can just use `#getPropertyValue`)